src/AppKit/NSMenuItem.cs

```diff
@@ -70,5 +70,14 @@ public NSMenuItem (string title) : this (title, null, "")
 			}
 		}
 
+		public Func <NSMenuItem, bool> ValidateMenuItem {
+			get {
+				return (target as ActionDispatcher)?.ValidateMenuItemFunc;
+			}
+			set {
+				(target as ActionDispatcher).ValidateMenuItemFunc = value;
```

Comment:
Could target ever be not an ActionDispatcher? Maybe a menu item we're getting from a 3rd party?

Do we need to type check and throw? Right now this will NRE if we hit this case (unlike get).

Reply:
It is possible for the target to be something different here.  I meant for it to be a ?.ValidateMenuitemFunc same as the getter.  Would we rather throw an exception instead?

Reply:
As a user I would be more than a little upset of my set operation was ignored because under the hood wasn't what I thought it was.

I lean pretty strongly towards throw, thoughts @rolfbjarne @dalexsoto ?

Reply:
I think throwing is better than ignoring, but we should throw something useful that explains the incompatibility, not a NRE.

Reply:
Agree, using NRE would be just puzzling for our end users, maybe InvalidOperationException and explain a little in the Message